cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/service/DraftHandler.java

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.datapipeline.service;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+import io.cdap.cdap.api.NamespaceSummary;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.metrics.Metrics;
+import io.cdap.cdap.api.service.http.AbstractSystemHttpServiceHandler;
+import io.cdap.cdap.api.service.http.HttpServiceRequest;
+import io.cdap.cdap.api.service.http.HttpServiceResponder;
+import io.cdap.cdap.api.service.http.SystemHttpServiceContext;
+import io.cdap.cdap.datapipeline.draft.CodedException;
+import io.cdap.cdap.datapipeline.draft.DraftId;
+import io.cdap.cdap.datapipeline.draft.DraftService;
+import io.cdap.cdap.datapipeline.draft.DraftStoreRequest;
+import io.cdap.cdap.datapipeline.draft.SortRequest;
+import io.cdap.cdap.etl.proto.v2.DataStreamsConfig;
+import io.cdap.cdap.etl.proto.v2.ETLBatchConfig;
+import io.cdap.cdap.etl.proto.v2.ETLConfig;
+import io.cdap.cdap.internal.io.SchemaTypeAdapter;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.nio.charset.StandardCharsets;
+import javax.annotation.Nullable;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+/**
+ * Handler of drafts
+ */
+public class DraftHandler extends AbstractSystemHttpServiceHandler {
+  private static final Gson GSON = new GsonBuilder()
+    .setPrettyPrinting()
+    .registerTypeAdapter(Schema.class, new SchemaTypeAdapter())
+    .create();
+
+  // Injected by CDAP
+  @SuppressWarnings("unused")
+  private Metrics metrics;
+
+  private DraftService draftService;
+
+  @Override
+  public void initialize(SystemHttpServiceContext context) throws Exception {
+    super.initialize(context);
+    this.draftService = new DraftService(context, this.metrics);
+  }
+
+  /**
+   * Returns a list of drafts associated with a namespace and the current user
+   */
+  @GET
+  @Path("v1/contexts/{context}/drafts")
```

Comment:
Does it make sense to have "v1/contexts/{context}" as a top-level `@Path` directive, e.g. like in `DatasetInstanceHandler`.
Also I see that usually path looks like `Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}`. Is context a different thing? I see that its referred  as a namespace below. On the other side ValidationHandler is using "/v1/contexts". It would be good to have the common part extracted as a constant similar to Gateway as it would be easier to bump version when we need to.

Reply:
Good note on the top-level path, I've moved the common section of the path to the top-level.

Regarding your question on context vs namespace, yes those are different. `context` in this URL refers to the different namespaces defined by the datapipeline app (the ones the user controls). The `namespace-id` you mentioned is the CDAP namespace that is used to manage all the apps that are deployed to the platform. In our case, this namespace is always `system` since we deploy the pipeline app in the CDAP system namespace but the `context` defaults to `default` since thats what the pipeline app creates out of the box. The full URL to access one of these APIs is:
`/v3/namespaces/system/apps/pipeline/services/studio/methods/v1/contexts/default/drafts/...`
so you can see both `namespace-id` and `context` are part of the path.

Reply:
Hmm it seems like the top-level path wasnt working (I was getting 404 errors). I pulled the API version into a global variable to make it easier to update in the future.